backend/question-service/db.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	"context"
+
+	"github.com/guregu/dynamo"
+	"github.com/jackc/pgx/v5"
+)
+
+type store interface {
+	createQuestion(Question) (QuestionId, error)
+	getQuestion(QuestionId) (Question, error)
+	getQuestions() ([]Question, error)
+	updateQuestion(Question) error
+	deleteQuestion(QuestionId) error
+}
+
+type questionStore struct {
+	descriptionTable dynamo.Table
+	pgxDb            *pgx.Conn
+}
+
+func NewQuestionStore(descriptionTable dynamo.Table, pgxDb *pgx.Conn) store {
+	return &questionStore{
+		descriptionTable: descriptionTable,
+		pgxDb:            pgxDb,
+	}
+}
+
+var queries = map[string]string{
+	"get":    "select * from questions where questionId=$1",
+	"getAll": "select * from questions",
+	"add":    "insert into questions (questionTitle, categories, difficulty, attempts, createdAt) values ($1, $2, $3, $4, $5) returning questionId",
+	"update": "update questions set questionTitle=$1, categories=$2, difficulty=$3, attempts=$4, createdAt=$5 where questionId=$6",
+	"delete": "delete from questions where questionId=$1",
```

Comment:
Can I ask why you use raw SQL string from querying instead of using an ORM?

Reply:
i didn't use an ORM as I was only interacting with 1 object type and didn't see the need the power (and bloat) that an ORM provides. I also personally prefer writing my own sql queries and don't find ORMs to provide a easier experience when developing. What do you think?

side note: The sql queries here are templatised so they shouldn't be vulnerable to sql injection attacks

Reply:
I personally prefer ORM since it makes your database agnostic (doesn't matter if you use postgres or some other relational database) but for this simple case, raw SQL should be fine.